src/instruction.rs

```diff
@@ -19,9 +19,8 @@ impl fmt::Debug for Instruction {
     }
 }
 
-pub(crate) type Handler =
-    fn(&mut VirtualMachine, *const Instruction, &mut dyn World) -> InstructionResult;
-pub(crate) type InstructionResult = Result<*const Instruction, ExecutionEnd>;
+pub(crate) type Handler = fn(&mut VirtualMachine, &mut dyn World) -> InstructionResult;
+pub(crate) type InstructionResult = Option<ExecutionEnd>;
```

Comment:
Bikeshedding: It's weird to see a type named "..Result" be an `Option<_>`. I'd rename it `InstructionOutput` or remove the type alias altogether.

Reply:
Maybe the name is bad.

What's even worse IMO is the Option. I want to change it to something like `Running | Stopped`.